feature-pack/src/main/resources/layers/standalone/elytron-tls/layer-spec.xml

```diff
@@ -0,0 +1,36 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+  ~ Copyright 2022 Red Hat, Inc.
+  ~
+  ~ Licensed under the Apache License, Version 2.0 (the "License");
+  ~ you may not use this file except in compliance with the License.
+  ~ You may obtain a copy of the License at
+  ~
+  ~   http://www.apache.org/licenses/LICENSE-2.0
+  ~
+  ~ Unless required by applicable law or agreed to in writing, software
+  ~ distributed under the License is distributed on an "AS IS" BASIS,
+  ~ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  ~ See the License for the specific language governing permissions and
+  ~ limitations under the License.
+  -->
+<layer-spec xmlns="urn:jboss:galleon:layer-spec:1.0" name="elytron-tls">
+    <dependencies>
+        <layer name="cdi"/>
```

Comment:
Only for the example in `dependency/`, so it can be removed

Reply:
Update, turns out the `cdi` layer is what adds Weld to the server, so it is needed. Found this from testing and [wildfly-grpc-feature-pack](https://github.com/wildfly-extras/wildfly-grpc-feature-pack/blob/main/feature-pack/src/main/resources/layers/standalone/grpc/layer-spec.xml)

Reply:
What is Weld being used for?

Reply:
It performs the dependency injection, which was being used by the `example/` code, and is called in the [SubsystemParsingTestCase](https://github.com/cam-rod/elytron-tls/blob/81ff3e82857ec74355ab4e6cb8813e3a3d8dbc42/subsystem/src/test/java/org/wildfly/extension/elytron/tls/subsystem/SubsystemParsingTestCase.java#L94). I'm checking against WFCORE and the gRPC project to see if it's needed in the latter case.

Reply:
I don't think Weld was needed in the original `SubsystemParsingTestCase` and I don't think it was part of `elytron/layer-spec.xml` so might be worth checking if that can be removed.

Reply:
Checked the code, it is not called by WFCORE, but the gRPC feature pack does also use it in [SubsystemTestCase](https://github.dev/wildfly-extras/wildfly-grpc-feature-pack/blob/038e76cf5c0f65081862a7caa4b0f15e6af7eb3c/subsystem/src/test/java/org/wildfly/extension/grpc/SubsystemTestCase.java#L44).

Seems the reason is because Arquillian needs `org.wildfly.security.manager`, so Weld loads it in. [From the gRPC subsystem](https://github.dev/wildfly-extras/wildfly-grpc-feature-pack/blob/038e76cf5c0f65081862a7caa4b0f15e6af7eb3c/subsystem/src/main/java/org/wildfly/extension/grpc/deployment/GrpcDependencyProcessor.java#L52)

Reply:
I think the need for the CDI dependency there is separate from the security manager line. For now, I think the Weld dependency could be removed here since it looks like it was only being used by the example code.

Reply:
Alright, updated some sections in the code and it now builds the testsuite properly, so I'll remove it.